### PR TITLE
CSHARP-2506: Reduce allocations in BsonClassMapSerializer.SerializeClass

### PR DESCRIPTION
**Pull request by dnickless**
Small change that removes a memory issue as described here: https://jira.mongodb.org/browse/CSHARP-2506

### CODE REVIEW

**Review by BorisDog — APPROVED** [review]
LGTM

File: src/MongoDB.Bson/Serialization/Serializers/BsonClassMapSerializer.cs [hunk 1]
```diff
@@ -576,15 +576,12 @@ private void SerializeClass(BsonSerializationContext context, BsonSerializationA
         {
             var bsonWriter = context.Writer;
 
-            var remainingMemberMaps = _classMap.AllMemberMaps.ToList();
-
```
Comment (rstam) [inline src/MongoDB.Bson/Serialization/Serializers/BsonClassMapSerializer.cs:580]: The removed variable is still referenced in commented out code below.

Remove the commented out code also after verifying that it is no longer needed.

**Review by rstam — CHANGES_REQUESTED** [review]

File: src/MongoDB.Bson/Serialization/Serializers/BsonClassMapSerializer.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Bson/Serialization/Serializers/BsonClassMapSerializer.cs:580]: Verified that AutoTimeStampMemberMap was never used. Removed the comment.

**Review by rstam — APPROVED** [review]
LGTM

**Comment by BorisDog** [comment]
[Pushed](https://github.com/mongodb/mongo-csharp-driver/commit/84ff74d8d62be06ead9e0e1580a549e66beec8dc)

**Pull request closed without merging**